libsolidity/formal/CHC.cpp

```diff
@@ -1597,7 +1615,8 @@ map<unsigned, vector<unsigned>> CHC::summaryCalls(CHCSolverInterface::CexGraph c
 		q.pop();
 
 		Predicate const* nodePred = Predicate::predicate(_graph.nodes.at(node).name);
-		if (nodePred->isSummary() && (_root == root || nodePred->isInternalCall() || nodePred->isExternalCall()))
+		Predicate const* rootPred = Predicate::predicate(_graph.nodes.at(root).name);
+		if (nodePred->isSummary() && (_root == root || nodePred->isInternalCall() || nodePred->isExternalCallTrusted() || nodePred->isExternalCallUntrusted() || rootPred->isExternalCallUntrusted()))
```

Comment:
this seems to work, but I haven't found a better way to describe this condition

Reply:
The idea here is the following: if a predicate has type `FunctionSummary`, we report it only if its summary parent is the error node, otherwise, inside the counterexample subgraph of a function, we're interested only in predicates of type `InternalCall` and `ExternalCall` (now `ExternalCallTrusted`). This was already the case for internal calls (the removed function in this diff).
However, now we're also interested in predicates of type `ExternalCallUntrusted` (the `nondet` predicates) and predicates of type `FunctionSummary` that have summary parent `ExternalCallUntrusted`. These are reentrant calls.